lib/impure/rdstdin.nim

```diff
@@ -100,12 +100,17 @@ when defined(Windows):
     stdout.write "\n"
 
 else:
-  import linenoise, termios
+  import linenoise, termios, posix
 
   proc readLineFromStdin*(prompt: string): TaintedString {.
                           tags: [ReadIOEffect, WriteIOEffect].} =
+    errno = 0
```

Comment:
https://linux.die.net/man/3/errno
> _errno_ is defined by the ISO C standard to be a modifiable lvalue of type _int_ [...]; _errno_ may be a macro
Therefore writing to it is just fine, even necessary to get correct error codes.